pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -814,3 +847,80 @@ def plot_channel_contributions_grid(
         )
         return fig
+
+    def _validate_data(self, X, y=None):
+        return X
+
+    def sample_posterior_predictive(
+        self,
+        X_pred,
+        extend_idata,
+        combined,
+        include_last_observations: bool = False,
+        original_scale: bool = True,
+        **kwargs,
+    ):
+        """
+        Sample from the model's posterior predictive distribution.
+
+        Parameters
+        ---------
+        X_pred : array, shape (n_pred, n_features)
+            The input data used for prediction using prior distribution..
+        extend_idata : Boolean determining whether the predictions should be added to inference data object.
+            Defaults to False.
+        combined: Combine chain and draw dims into sample. Won't work if a dim named sample already exists.
+            Defaults to True.
+        include_last_observations: Boolean determining whether to include the last observations of the training data.
+            Defaults to False.
+        original_scale: Boolean determining whether to return the predictions in the original scale of the target variable.
+            Defaults to True.
+        **kwargs: Additional arguments to pass to pymc.sample_posterior_predictive
+
+        Returns
+        -------
+        posterior_predictive_samples : DataArray, shape (n_pred, samples)
+            Posterior predictive samples for each input X_pred
+        """
+        if include_last_observations:
+            X_pred = pd.concat(
+                [self.X.iloc[-self.adstock_max_lag :, :], X_pred], axis=0
+            )
+
+        self._data_setter(X_pred)
+
+        with self.model:  # sample with new input data
+            post_pred = pm.sample_posterior_predictive(self.idata, **kwargs)
+            if extend_idata:
+                self.idata.extend(post_pred, join="right")  # type: ignore
+
+        posterior_predictive_samples = az.extract(
+            post_pred, "posterior_predictive", combined=combined
+        )
+
+        if include_last_observations:
+            posterior_predictive_samples = posterior_predictive_samples.isel(
+                date=slice(self.adstock_max_lag, None)
+            )
+
+        if original_scale:
+            # These are lost during the ufunc
+            attrs = posterior_predictive_samples.attrs
+
+            if combined:
+                posterior_predictive_samples = xr.apply_ufunc(
+                    self.get_target_transformer().inverse_transform,
+                    posterior_predictive_samples,
+                )
+            else:
+                posterior_predictive_samples = xr.apply_ufunc(
+                    self.get_target_transformer().inverse_transform,
+                    posterior_predictive_samples.expand_dims(dim={"_": 1}, axis=1),
+                    input_core_dims=[["date", "_"]],
+                    output_core_dims=[["date", "_"]],
+                    vectorize=True,
+                ).squeeze(dim="_")
```

Comment:
I understand this because I wrote it but it might be confusing. So you could write a little helper function to wrap this logic and add a unit-test (just a suggestion)

Reply:
Will do!